### PR TITLE
fix(cxx_indexer): insist that compilation units have working directories

### PR DESCRIPTION
**Pull request by zrlk**
The C++ (+ friends) indexer should be hermetic modulo claiming when
reading from extracted compilation units. Currently, it will use the
indexing environment's working directory when one is not set in the
compilation unit (or if the value that was set is not an absolute path).
This PR turns this into an error condition.

### CODE REVIEW

File: kythe/cxx/indexer/cxx/frontend.cc [hunk 1]
```diff
@@ -123,11 +123,10 @@ void MaybeNormalizeFileVNames(IndexerJob* job) {
...
 void UpdateJobWdirFromUnit(IndexerJob* job) {
   job->working_directory = job->unit.working_directory();
-  if (!llvm::sys::path::is_absolute(job->working_directory)) {
-    llvm::SmallString<1024> stored_wd;
-    CHECK(!llvm::sys::fs::make_absolute(stored_wd));
-    job->working_directory = stored_wd.str();
-  }
+  CHECK(!job->working_directory.empty())
+      << "Indexer jobs must have their absolute working directory set.";
+  CHECK(llvm::sys::path::is_absolute(job->working_directory))
```
Comment (creachadair) [inline kythe/cxx/indexer/cxx/frontend.cc]: Does this not imply the previous condition already?

Comment (zrlk) [reply kythe/cxx/indexer/cxx/frontend.cc]: Yes, but I expect custom extractors to forget to populate the field entirely more often than to populate it incorrectly.

Comment (creachadair) [reply kythe/cxx/indexer/cxx/frontend.cc]: Well, okay, but then should that case have a different diagnostic?

Comment (creachadair) [reply kythe/cxx/indexer/cxx/frontend.cc]: The other problem with this solution is that the indexer will now fail on stored compilations that didn't set this value. Maybe instead we should assume "/" if it's not set, and only issue a diagnostic?

**Pull request merged**